motioneye/mjpgclient.py

```diff
@@ -278,53 +288,50 @@ def _seek_content_length(self):
         r_future.add_done_callback(self._on_before_content_length)
 
     def _on_before_content_length(self, future: Future):
-        try:
-            future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            if self._check_error():
-                return
+        result, _ = self._get_future_result(future)
+        if not result:
+            return
+
+        if self._check_error():
+            return
 
-            r_future = utils.cast_future(self.read_until(b'\r\n\r\n'))
-            r_future.add_done_callback(self._on_content_length)
+        r_future = utils.cast_future(self.read_until(b'\r\n\r\n'))
+        r_future.add_done_callback(self._on_content_length)
 
     def _on_content_length(self, future: Future):
-        try:
-            data = future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            if self._check_error():
-                return
-
-            matches = re.findall(rb'(\d+)', data)
-            if not matches:
-                self._error(
-                    'could not find content length in mjpg header line "{header}"'.format(
-                        header=data
-                    )
+        result, data = self._get_future_result(future)
+        if not result:
+            return
+
+        if self._check_error():
+            return
+
+        matches = re.findall(rb'(\d+)', data)
+        if not matches:
+            self._error(
+                'could not find content length in mjpg header line "{header}"'.format(
+                    header=data
                 )
+            )
 
-                return
+            return
 
-            length = int(matches[0])
+        length = int(matches[0])
 
-            r_future = utils.cast_future(self.read_bytes(length))
-            r_future.add_done_callback(self._on_jpg)
+        r_future = utils.cast_future(self.read_bytes(length))
+        r_future.add_done_callback(self._on_jpg)
 
     def _on_jpg(self, future: Future):
-        try:
-            data = future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            self._last_jpg = data
-            self._last_jpg_times.append(time.time())
-            while len(self._last_jpg_times) > self._FPS_LEN:
-                self._last_jpg_times.pop(0)
+        result, data = self._get_future_result(future)
+        if not result:
+            return
```

Comment:
The error check wasn't there before, but I'm sure the flag is set in `_get_future_result()`, hence it makes sense to return here as well if it is true?
```suggestion
        if not result or self._check_error():
            return
```

Reply:
Ah you are right, I see the `_check_error` call in `_seek_content_length`. Whether or not it makes sense to assign `_last_jpg = data` in case of an error is a different question, but I guess `result` is mostly empty in this case as well. Let's leave it as it is.